tests/devapps/DesktopTestApp/PublicClientHandler.cs

```diff
@@ -38,11 +40,11 @@ class PublicClientHandler
         public PublicClientHandler(string clientId)
         {
             ApplicationId = clientId;
-            PublicClientApplication = new PublicClientApplication(ApplicationId)
-            {
-                UserTokenCache = TokenCacheHelper.GetUserCache(),
-                Component = _component
-            };
+            PublicClientApplication = PublicClientApplicationBuilder.Create(ApplicationId)
+                .WithComponent(_component)
```

Comment:
>WithComponent [](start = 17, length = 13)

What is WithComponent?

Reply:
This is the component name that apps can set to differentiate themselves in telemetry.  It's a terrible name, but it's currently a mirror for the Component property on clientapplicationbase.  We should review this as part of the updated public api surface.

---
In reply to: [247743496](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/786#discussion_r247743496) [](ancestors = 247743496)